src/util/std_types.h

```diff
@@ -126,39 +110,36 @@ class symbol_typet:public typet
   }
 };
 
-/*! \brief Cast a generic typet to a \ref symbol_typet
- *
- * This is an unchecked conversion. \a type must be known to be \ref
- * symbol_typet.
- *
- * \param type Source type
- * \return Object of type \ref symbol_typet
- *
- * \ingroup gr_std_types
-*/
+/// \brief Cast a generic typet to a \ref symbol_typet.
```

Comment:
Though you are just moving code here, the use of the word generic tripped me up since a "generic type" normally means `List<Integer>`! I think `Cast a typet to a symbol_typet` is actually less ambiguous.